L1Trigger/L1TGlobal/src/GlobalBoard.cc

```diff
@@ -1180,23 +1182,45 @@ const std::vector<double> l1t::GlobalBoard::semirandomNumber(const edm::Event& i
   std::srand(std::rand() + iEvent.id().event());
   // very large (semi)random number
   double const semirandom = std::rand();
-  for (auto& ps : out) {
-    // if the ps is smaller than 1 (e.g. ps=0, ps=1), it is not changed
-    // else, replace ps with a semirandom integer in the [1,ps] range
-    if (ps > 1) {
-      auto nps = semirandom - floor(semirandom / ps) * ps;
-      // if nps=0 or a wrong value (<0,>ps) use PS value (standard method)
-      if (nps > 0 and nps <= ps)
-        ps = nps;
-      else {
-        if (nps != 0)  // complain only if nps <0 or nps >PS
-          edm::LogWarning("L1TGlobal::semirandomNumber")
-              << "\n The inital prescale counter obtained by L1TGlobal::semirandomNumber is wrong."
-              << "\n This is probably do to the floating-point precision. Using the PS value."
-              << "\n semirandom = " << semirandom << "\n PS = " << ps << "\n nps = " << nps
-              << " <-- it should be in the range [0 , " << ps << "]" << std::endl;
-      }
+
+  for (size_t iAlgo = 0; iAlgo < prescaleFactorsAlgoTrig.size(); iAlgo++) {
+    int ps = std::lround(prescaleFactorsAlgoTrig[iAlgo] * std::pow(10, m_precision));
+    int nps = semirandom - floor(semirandom / ps) * ps;
+
+    // if nps=ps or a wrong value (<0,>ps) use zero
+    if (nps >= 0 and nps < ps)
+      out.push_back(PrescaleCounter(prescaleFactorsAlgoTrig[iAlgo], m_precision, nps));
+    else {
+      edm::LogWarning("L1TGlobal::semirandomNumber")
```

Comment:
Since L1174 was changed (`auto out = prescaleFactorsAlgoTrig;`), in the new L1194, we get a warning, but no element is added to the vector, so the vector size won't be the same as that of the input vector, and that seems wrong.